src/llir/abi.rs

```diff
@@ -424,7 +426,7 @@ fn abi_to_signature(abi: &InstrAbi, abi_span: Span, ctx: &mut CompilerContext<'_
 
     defs::Signature {
         return_ty: sp!(value::ExprType::Void),
-        params: abi.encodings.iter().enumerate().flat_map(|(index, enc)| {
+        params: abi.encodings.iter().filter(|enc| !matches!(*enc, ArgEncoding::Padding { .. })).enumerate().flat_map(|(index, enc)| {
```

Comment:
I was concerned something like that would end up inserting a None into the resulting collection. Didn't know it would skip over it. Seems like there could be issues in the future though if `index` is ever used for anything though.

Reply:
Oops I missed the fact that this would change indices, lemme read this over again.

Reply:
> I was concerned something like that would end up inserting a None into the resulting collection. 

`flat_map` flattens the iterators being returned by the closures.  `Option` is an iterator of zero or one items, so `None` produces no items.  Technically, this should be using `.filter_map` instead; this is an alternative name for a `flat_map` to `Option` that is a bit clearer in its purpose.

> Oops I missed the fact that this would change indices, lemme read this over again.

So the index is only used to generate arg names.  I would pull that out.

```rust
// in a line of code outside the loop
let arg_names = (1..).map(|i| ident!("arg_{i}"));

// inside the loop, change this line:
let name = sp!(abi_span => ctx.resolutions.attach_fresh_res(arg_names.next().unwrap()));
```

Reply:
and of course get rid of the `enumerate()`

Reply:
Would it make more sense to just add another function to the `InstrAbi` implementation similar to `arg_encodings` that returns the filtered version? This seems like the sort of thing that would end up continually popping up throughout the code base otherwise. (I would've done that in the first place, but rust complained about the filter output not being compatible with `VeclikeIterator` or something and I don't know how to make it happy.)

Reply:
Hmmmmm.  Crap, that's a pain.

Writing this function is probably a good idea.  It will have to return the slightly weaker `DoubleEndedIterator` instead of `VeclikeIterator`.  (basically because `filter` can't be `ExactSizeIterator`, which is the other half of `VeclikeIterator`)